### PR TITLE
new WORKING machine - Mortal Race

### PR DESCRIPTION
**Pull request by DavidHaywood**
new WORKING machines
--------
Mortal Race [Museo del Recreativo, David Haywood, ClawGrip]

### CODE REVIEW

File: src/mame/edevices/stlforce.cpp [hunk 1]
```diff
@@ -219,21 +230,33 @@ static INPUT_PORTS_START( stlforce )
...
 INPUT_PORTS_END
 
+static const gfx_layout stlforce_bg_full_layout =
+{
+	16,16,
+	RGN_FRAC(1,1),
+	4,
+	{0,1,2,3},
+	{12,8,4,0,28,24,20,16,16*32+12,16*32+8,16*32+4,16*32+0,16*32+28,16*32+24,16*32+20,16*32+16},
+	{0*32,1*32,2*32,3*32,4*32,5*32,6*32,7*32,8*32,9*32,10*32,11*32,12*32,13*32,14*32,15*32},
```
[the remark this answers is not in the document]
Comment (MooglyGuy) [reply src/mame/edevices/stlforce.cpp]: Out of curiosity, is there any practical documentation when it comes to these sorts of things? I just had a look at docs.mamedev.org, and unless I'm not looking at the right pages, documentation on the magical incantations involved with `gfxdecode` seems pretty thin on the ground, and documentation on the helper macros even less so.

[the remark this answers is not in the document]
Comment (angelosa) [reply src/mame/edevices/stlforce.cpp]: It's pretty much as opaque as it gets:
```python
gfxdecode = {
   x_char_size, y_char_size,
   allocated_char_size,
   bpp_number,
   { bitplane_offset_start0, ... , bitplane_offset_startN },
   { x_position0, ..., x_positionN },
   { y_position0, ..., y_positionN }
 }
```

- X and Y positions counts in bits, iirc left to right;
- `RGN_FRAC(offset, length)` applies to the region, divides by length and starts at offset.
- `STEPN(base, adder)` again in bits, starts from the base bit and adds adder for every N.

```python
	GFXDECODE_ENTRY( "gfx2", 0, char_layout, 0, 64 )
```

In order: region_name, tile_offset_start, name_of_fn, start_offset_palette (iirc the base GLOBAL color entry), number_of_palette_entries

In short:
- it's pretty much C code of a broken data structure;
- making it to cooperate is pretty much trial & error, I hate every second I have to mess with it;
- there's no way to edit the structures from debugger or the F4 menu. You either be ready with a DIY tool or just pay the compiling time overhead;
- there's no real check to make sure it doesn't go out of bounds with colors and tiles;
